codex/stores/blockstore.nim

```diff
@@ -50,7 +48,7 @@ method getBlock*(self: BlockStore, cid: Cid): Future[?!Block] {.base.} =
 method putBlock*(
   self: BlockStore,
   blk: Block,
-  ttl = Duration.none): Future[?!void] {.base.} =
+  ttl = times.Duration.none): Future[?!void] {.base.} =
```

Comment:
I chose not to use chronos' Duration or Moment for this use-case. It uses an int64 to represent time in nano-seconds. I remember figuring out that using this for timestamps would have us run out of bits some time in the 2040s. I checked just now, and it seem we wouldn't run out until 2262. Perhaps you could check? Mark's been using a SecondsSince1970 type as an int64 and I kind of want to use that. It's easy to understand and definitely future-proof.

I saw the Duration and Moment from chronos were definitely designed to run with its high-speed timer code. Its largest supported time constant (nim-chronos/chronos/timer.nim line 298) is Week.

Reply:
I wasn under the impression we were mixing chronos and std timers, it appears that we aren't, which is good. Using epoch here actually makes sence since we're simply recording it in the store.

Reply:
Well, your impression is partially right: I am using both chronos and std/times Duration types. I need to use chronos for the asyncSleep method. And if I only use chronos then I have to convert seconds into nanoseconds, and then back again for no reason. I don't understand why chronos introduces its own Duration type. The std/times one seems to do everything the chronos one does.
So, the way I see it:
A) Use chronos in both cases and do unnecessary unit conversion
B) Use chronos for the timer duration and std/times for the blockTTL duration
I don't really like either option. What do you think?

Reply:
So `chronos` has it's own functions for epoch:

```Nim
func epochSeconds*(moment: Moment): int64 =
  moment.value div Second.value

func epochNanoSeconds*(moment: Moment): int64 =
  moment.value
```

We want to stick with `chronos` because we're heavily dependent on it across the stack. Check out the `chronos.timer` module, specially the `Moment.now`, `Moment.init` and epoch methods. 

My approach with the `ZeroMoment` was based on an older version of `chronos` that didn't expose the epoch methods, which this, the systems times module should not be necessary anymore.

Reply:
I've removed std/times and used chronos for Durations everywhere. And I'm using SecondsSince1970 for timestamps. This gives a conversion overhead, but it's not a problem and it does clean up some code.